accord-core/src/main/java/accord/coordinate/Infer.java

```diff
@@ -168,18 +170,24 @@ public static Status invalidIfNotAtLeast(SafeCommandStore safeStore, TxnId txnId
         return Status.NotDefined;
     }
 
-    public static boolean inferInvalidated(CheckStatus.WithQuorum withQuorum, Status invalidIfNotAtLeast, SaveStatus saveStatus, SaveStatus maxSaveStatus)
+    public static boolean inferInvalidated(CheckStatus.WithQuorum withQuorum, Status minInvalidIfNotAtLeast, Status maxInvalidIfNotAtLeast, SaveStatus saveStatus, SaveStatus maxSaveStatus, Ballot minPromised)
     {
         if (saveStatus == SaveStatus.Invalidated)
             return true;
 
         if (withQuorum != HasQuorum)
             return false;
 
-        // should not be possible to reach a quorum without finding the definition unless cleanup is in progress
-        Invariants.checkState(saveStatus != SaveStatus.Accepted || maxSaveStatus.phase == Status.Phase.Cleanup);
+        // if everyone has witnessed a recovery, or everyone had invalidIfNotAtLeast, then we can rely on any inference;
+        // otherwise we might have raced with the original coordinator, and only witnessed part of their effects
+        Status invalidIfNotAtLeast = minPromised.compareTo(Ballot.ZERO) > 0 ? maxInvalidIfNotAtLeast : minInvalidIfNotAtLeast;
```

Comment:
I have updated this logic, but the basic premise is: if we have witnessed a quorum where _everyone_ has promised some non-zero ballot, then we know for sure that the original coordinator may not take some action after us (because every quorum it could obtain after us would reject it)